frontend/src/pages/FindChain.tsx

```diff
@@ -84,32 +95,51 @@ const FindChain = () => {
     })();
   }, []);
 
-  const handleChange = (e: { target: { checked: boolean; value: any } }) => {
-    if (e.target.checked) {
-      setGender([...gender, e.target.value]);
+  //get selected categories
+  const handleChange = (
+    e: any,
+    value: any,
+    categories: any,
+    setCategories: any
+  ) => {
+    if (!categories.includes(value)) {
+      setCategories([...categories, value]);
     } else {
-      setGender(gender.filter((id) => id !== e.target.value));
+      setCategories(categories.filter((id: any) => id !== value));
     }
   };
 
+  //filter selected categories
   useEffect(() => {
-    if (gender.length === 0) {
-      setFilteredChains(chainData);
-      return;
-    }
+    const findCommonElementsArrays = (arr1: any, arr2: any) =>
```

Comment:
I would call this "hasCommonElements". You're returning a bool, so not finding anything.